File: php/src/CarrierAddressClient.php
<?php

namespace Zymeup\SDK;

class CarrierAddressClient
{
    private ZymeupClient $client;

    public function __construct(ZymeupClient $client)
    {
        $this->client = $client;
    }

    public function list(array $params = []): array
    {
        return $this->client->request('POST', '/api/v1/carrier/sdk/addresses/list', $params);
    }

    public function create(array $data): array
    {
        return $this->client->request('POST', '/api/v1/carrier/sdk/addresses/create', $data);
    }

    public function update(string $id, array $data): array
    {
        return $this->client->request('POST', "/api/v1/carrier/sdk/addresses/{$id}/update", $data);
    }

    public function delete(string $id): array
    {
        return $this->client->request('POST', "/api/v1/carrier/sdk/addresses/{$id}/delete", []);
    }

    public function setDefault(string $id): array
    {
        return $this->client->request('POST', "/api/v1/carrier/sdk/addresses/{$id}/set-default", []);
    }
}

File: swift/Sources/ShipzySDK/CarrierAddressClient.swift
import Foundation

// ============ Carrier Address Client ============

public class CarrierAddressClient: HttpClient {
    public func list(_ params: [String: Any] = [:]) async throws -> TenantAddressListResponse {
        let body = try JSONSerialization.data(withJSONObject: params)
        return try await request("/api/v1/carrier/sdk/addresses/list", method: "POST", body: body)
    }

    public func create(_ data: [String: Any]) async throws -> TenantAddress {
        let body = try JSONSerialization.data(withJSONObject: data)
        return try await request("/api/v1/carrier/sdk/addresses/create", method: "POST", body: body)
    }

    public func update(_ id: String, _ data: [String: Any]) async throws -> TenantAddress {
        let body = try JSONSerialization.data(withJSONObject: data)
        return try await request("/api/v1/carrier/sdk/addresses/\(id)/update", method: "POST", body: body)
    }

    public func delete(_ id: String) async throws -> [String: Bool] {
        return try await request("/api/v1/carrier/sdk/addresses/\(id)/delete", method: "POST", body: Data())
    }

    public func setDefault(_ id: String) async throws -> TenantAddress {
        return try await request("/api/v1/carrier/sdk/addresses/\(id)/set-default", method: "POST", body: Data())
    }
}
